helix-core/src/main/java/org/apache/helix/task/AbstractTaskDispatcher.java

```diff
@@ -745,7 +743,7 @@ protected void scheduleForNextTask(String job, JobContext jobCtx, long now) {
     }
   }
 
-  // add all partitions that have been tried maxNumberAttempts
+  // add all partitions that are given up
```

Comment:
What is the distinction between "given up" and partitions that exceeded the # of maximum attempts?

Reply:
Thanks for the comment. I added comments for more explanation. 
Also I noticed the given up tasks has been already added in job dispatcher. So we do not need to add the tasks again. However, we want to have a sanity check to filter all of the tasks, regarding of their states, if they reached their maximum number of attempts. Because it has been observed that sometimes, the state of the task might be INIT or DROPPED or other states (if we have pending messages) and we miss max number of attempts check for those tasks.